viewing_party/party.py

```diff
@@ -1,23 +1,153 @@
+    if watched == []:
+        return 0.0
+    # sums the ratings of all the movies that have been watched. returns average rating.
+    total = 0
+    for movie in watched:
+        total += movie["rating"]
+
+    return total / len(watched)
+
+def get_most_watched_genre(user_data):
+    # If the user hasn't watched any movies, return None
+    if not user_data["watched"]:
+        return None
+
+    genre_count = {}
+
+    # Loop through watched movies and get genre of current movie
+    for movie in user_data["watched"]:
+        genre = movie["genre"] 
+        
+        if genre in genre_count:
+            genre_count[genre] += 1
+
+        else:
+            genre_count[genre] = 1
+
+    # return genre with highest count of movies
+    most_watched = max(genre_count, key=genre_count.get)
+    return most_watched
+
+    
+# get list of movies user has watched, but friends haven't.
+def get_unique_watched(user_data):
+    unique_movies = []
+
+    friends_watched = []
+    for friend in user_data["friends"]:
+        for movie in friend["watched"]:
+            if movie not in friends_watched:
+                friends_watched.append(movie)
 
-# -----------------------------------------
-# ------------- WAVE 2 --------------------
-# -----------------------------------------
+    for movie in user_data["watched"]:
+        if movie not in friends_watched:
+            unique_movies.append(movie)
 
+    return unique_movies
 
-# -----------------------------------------
-# ------------- WAVE 3 --------------------
-# -----------------------------------------
 
+def get_friends_unique_watched(user_data):
+    friends_movie_for_user = []
+
+    user_watched = set()
+    for movie in user_data["watched"]:
+        user_watched.add(movie["title"])
+
+    friends_watched = set()
+    for friend in user_data["friends"]:
+        for movie in friend["watched"]:
+            if movie["title"] not in user_watched and movie["title"] not in friends_watched:
+                friends_movie_for_user.append(movie)
+                friends_watched.add(movie["title"])
+
+    return friends_movie_for_user
+        
         
-# -----------------------------------------
-# ------------- WAVE 4 --------------------
-# -----------------------------------------
+def get_available_recs(user_data):
+    recommended_list = []
+    subscriptions = set(user_data.get("subscriptions", []))
+    user_watched_titles = {movie["title"] for movie in user_data.get("watched", [])}
+    recommended_titles = set()
+
+    for friend in user_data.get("friends", []):
+        for movie in friend.get("watched", []):
+            if (
+                movie["title"] not in user_watched_titles
+                and movie["host"] in subscriptions
+                and movie["title"] not in recommended_titles
+            ):
+                recommended_list.append(movie)
+                recommended_titles.add(movie["title"])
+
+    return recommended_list

```

Comment:
The significant difference between this function `get_available_recs` and the prior function `get_friends_unique_watched` is that here, `get_available_recs` also needs to check if the host service of a movie is in the user's subscriptions.

Since we already did the work to create a function that returns the movies we have not seen but at least one friend has, how could we use the result of our existing function to simplify `get_available_recs` and remove any duplicate work the code's performing?